dmc/configs/fig7_flops.py

```diff
@@ -42,33 +45,37 @@
 
 
 def update_x_percent_threshold_in_config(
-    config,
-    x_percent_threshold,
-    evidence_update_threshold="x_percent_threshold",
-):
+    template: dict,
+    x_percent_threshold: int,
+) -> dict:
     """Update the x_percent threshold in the config.
     This function modifies the config in-place.
 
     Args:
-        config (dict): The config to update.
-        x_percent_threshold (float): The percentage of the threshold to update.
-        evidence_update_threshold (str): How to decide which hypotheses should be updated.
-            In [int, float, 'mean', 'median', 'all', 'x_percent_threshold'].
+        template (dict): The config to update.
+        x_percent_threshold (int): The percentage of the threshold to update.
 
     Returns:
         dict: The updated config.
     """
+    config = copy.deepcopy(template)
 
     # Update the x_percent_threshold
     lm_config_dict = config["monty_config"].learning_module_configs
     lm_config_dict["learning_module_0"]["learning_module_args"][
         "x_percent_threshold"
     ] = x_percent_threshold
 
-    # Update the string value for evidence_update_threshold
+    # Explicitly set the evidence update to "80%"
     lm_config_dict["learning_module_0"]["learning_module_args"][
         "evidence_update_threshold"
-    ] = evidence_update_threshold
+    ] = "80%"
```

Comment:
It might be clearer to have this as a default value for `update_x_percent_threshold_in_config`, and set it based on that (in which case worth updating the comment on line 69).